Detectors/PHOS/workflow/src/ClusterizerSpec.cxx

```diff
@@ -53,36 +52,17 @@ void ClusterizerSpec::run(framework::ProcessingContext& ctx)
 
     LOG(DEBUG) << "PHOSClusterizer - run run on cells called";
 
-    // auto dataref = ctx.inputs().get("cells");
-    // auto const* phosheader = o2::framework::DataRefUtils::getHeader<o2::phos::PHOSBlockHeader*>(dataref);
-    // if (!phosheader->mHasPayload) {
-    //   LOG(DEBUG) << "[PHOSClusterizer - run] No more cells" << std::endl;
-    //   ctx.services().get<o2::framework::ControlService>().readyToQuit(framework::QuitRequest::Me);
-    //   return;
-    // }
-
     auto cells = ctx.inputs().get<gsl::span<o2::phos::Cell>>("cells");
     LOG(DEBUG) << "[PHOSClusterizer - run]  Received " << cells.size() << " cells, running clusterizer ...";
     auto cellsTR = ctx.inputs().get<gsl::span<o2::phos::TriggerRecord>>("cellTriggerRecords");
 
-    printf("-------------Cells------------\n");
-    for (auto cTR : cellsTR) {
-      printf("new Event \n");
-      int firstCellInEvent = cTR.getFirstEntry();
-      int lastCellInEvent = firstCellInEvent + cTR.getNumberOfObjects();
-      for (int i = firstCellInEvent; i < lastCellInEvent; i++) {
-        Cell c = cells[i];
-        printf("cell[%d]: absId=%d, amp=%f, Time=%e \n", i, c.getAbsId(), c.getEnergy(), c.getTime());
-      }
+    std::unique_ptr<const o2::dataformats::MCTruthContainer<o2::phos::MCLabel>> truthcont;
```

Comment:
Dear @shahor02, do we copy list of primaries here or just move pointer? Is it correct way to access primaries?

Reply:
Hi @peressounko Yes, this is correct, accessing MCTruthContainer from the input will create internally a copy, you just access it via ``unique_ptr``. There is also a copy-less method, which requires different way of sending the message:
https://github.com/AliceO2Group/AliceO2/blob/86f40deaceafe12d7638abd78679c19213457a25/Steer/DigitizerWorkflow/src/ITSMFTDigitizerSpec.cxx#L180-L184

writing:
https://github.com/AliceO2Group/AliceO2/blob/86f40deaceafe12d7638abd78679c19213457a25/Detectors/ITSMFT/common/workflow/src/DigitWriterSpec.cxx#L85-L87

and reading:
https://github.com/AliceO2Group/AliceO2/blob/86f40deaceafe12d7638abd78679c19213457a25/Detectors/ITSMFT/ITS/workflow/src/ClustererSpec.cxx#L83-L88

Eventually, we should switch to these methods in all workflows.

Reply:
Thanks, I will try in next commit.